File: assignment1/leg.py
class Leg:
    DEFAULT_LENGTH : float = 20.0 # inches
    MAX_WEIGHT : float = 75 # pounds per leg

    def __init__(self, length : float = DEFAULT_LENGTH):
        self.length = length
        self.broken = False
    
    def getLength(self) -> float:
        return self.length

    def isBroken(self) -> bool:
        return self.broken
    
    def setBroken(self, value : bool) -> None:
        self.broken = value

File: assignment2/src/leg.cpp
#include <stdexcept>
#include "leg.h"

Leg::Leg(float length) {
    Leg::_length = length;
    Leg::_broken = false;
    if(length < 0)
        throw std::invalid_argument("Length must be > 0");
}

float Leg::getLength() const { return _length; }
bool Leg::isBroken() const { return _broken; } 

void Leg::setBroken(bool broken) {
    Leg::_broken = broken;
}
